src/utilities/util_dump.c

```diff
@@ -805,15 +842,328 @@ dump_all_records(WT_CURSOR *cursor, bool reverse, bool json)
+                            printf("#%d: %s\n", i, bookmarks[i]);
+                    }
+                }
+                /* Jump to the bookmark. */
+                else {
+                    if (util_str2num(session, args[1], true, &bookmark_index) != 0 ||
+                      bookmark_index >= MAX_BOOKMARKS) {
+                        printf("Error: please indicate a value between 0 and %d\n", MAX_BOOKMARKS);
+                        break;
+                    }
+                    key = bookmarks[bookmark_index];
+                    if (key == NULL) {
+                        printf(
+                          "Error: no keys associated with bookmark %" PRIu64 ".\n", bookmark_index);
+                        break;
+                    }
+
+                    /* Set the cursor to the bookmark. */
+                    cursor->set_key(cursor, key);
+                    ret = cursor->search(cursor);
+                    if (ret != 0 && ret != WT_NOTFOUND)
+                        return (util_cerr(cursor, "search", ret));
+                    else if (ret == WT_NOTFOUND) {
+                        printf("Error: %d\n", ret);
+                        ret = 0;
+                    } else
+                        printf("Cursor positioned on key %s.\n", key);
+                }
+            }
+            /* Delete. */
+            else if (strcmp(first_arg, "bd") == 0) {
+                if (num_args < 2) {
+                    printf("Error: please indicate the bookmark you want to delete.\n");
+                    break;
+                }
+                if (util_str2num(session, args[1], true, &bookmark_index) != 0 ||
+                  bookmark_index >= MAX_BOOKMARKS)
+                    printf("Error: please indicate a value between 0 and %d\n", MAX_BOOKMARKS);
+                else {
+                    free(bookmarks[bookmark_index]);
+                    bookmarks[bookmark_index] = NULL;
+                    printf("Bookmark %" PRIu64 " deleted.\n", bookmark_index);
+                }
+            }
+            /* Save. */
+            else if (strcmp(first_arg, "bs") == 0) {
+                if (num_args >= 2) {
+                    key = args[1];
+                    cursor->set_key(cursor, key);
+                    ret = cursor->search(cursor);
+                    if (ret != 0)
+                        return (util_cerr(cursor, "search", ret));
+                }
+                ret = cursor->get_key(cursor, &key);
+                if (ret != 0 && ret != EINVAL)
+                    return (util_cerr(cursor, "get_key", ret));
+                else if (ret == EINVAL) {
+                    printf("Error: the cursor needs to be positioned to create a bookmark.\n");
+                    ret = 0;
+                    break;
+                }
+                for (i = 0; i < MAX_BOOKMARKS; ++i) {
+                    if (bookmarks[i] == NULL) {
+                        if ((bookmarks[i] = malloc(strlen(key))) == NULL)
```

Comment:
This leaks memory when the function exits. While I realize the util itself is probably going to exit. Still I suggest cleaning up the memory or providing a comment as to why it doesn't matter.

Reply:
Updated in 3a8666c193a870d2a071dd349105c98635176418, on failure we will jump to the `err` label that will do the cleanup.